src/nix.rs

```diff
@@ -0,0 +1,338 @@
+            )
+            .into(),
+            Div => Term::Op2(
+                BinaryOp::Div(),
+                translate(lhs, file_id),
+                translate(rhs, file_id),
+            )
+            .into(),
+
+            Equal => Term::Op2(
+                BinaryOp::Eq(),
+                translate(lhs, file_id),
+                translate(rhs, file_id),
+            )
+            .into(),
+            Less => Term::Op2(
+                BinaryOp::LessThan(),
+                translate(lhs, file_id),
+                translate(rhs, file_id),
+            )
+            .into(),
+            More => Term::Op2(
+                BinaryOp::GreaterThan(),
+                translate(lhs, file_id),
+                translate(rhs, file_id),
+            )
+            .into(),
+            LessOrEq => Term::Op2(
+                BinaryOp::LessOrEq(),
+                translate(lhs, file_id),
+                translate(rhs, file_id),
+            )
+            .into(),
+            MoreOrEq => Term::Op2(
+                BinaryOp::GreaterOrEq(),
+                translate(lhs, file_id),
+                translate(rhs, file_id),
+            )
+            .into(),
+            NotEqual => Term::Op1(
+                UnaryOp::BoolNot(),
+                Term::Op2(
+                    BinaryOp::Eq(),
+                    translate(lhs, file_id),
+                    translate(rhs, file_id),
+                )
+                .into(),
+            )
+            .into(),
+
+            Implication => Term::App(
+                Term::App(
+                    Term::Op1(UnaryOp::Ite(), translate(lhs, file_id)).into(),
+                    translate(rhs, file_id),
+                )
+                .into(),
+                Term::Bool(true).into(),
+            )
+            .into(),
+
+            And => Term::App(
+                Term::Op1(UnaryOp::BoolAnd(), translate(lhs, file_id)).into(),
+                translate(rhs, file_id),
+            )
+            .into(),
+            Or => Term::App(
+                Term::Op1(UnaryOp::BoolOr(), translate(lhs, file_id)).into(),
+                translate(rhs, file_id),
+            )
+            .into(),
+        }
+    }
+}
+
+fn translate(node: rnix::SyntaxNode, file_id: FileId) -> RichTerm {
+    use rnix::types::{self, ParsedType, Wrapper};
+    use rnix::value;
+    use rnix::SyntaxKind::*;
+    use std::convert::{TryFrom, TryInto};
+    let pos = node.text_range();
+    let span = mk_span(file_id, pos.start().into(), pos.end().into());
+    println!("{:?}", node);
+    let node: ParsedType = node.try_into().unwrap();
+    match node {
+        ParsedType::Error(_) => Term::ParseError.into(),
+        ParsedType::Root(n) => translate(n.inner().unwrap(), file_id),
+        ParsedType::Paren(n) => translate(n.inner().unwrap(), file_id),
+        ParsedType::Dynamic(n) => translate(n.inner().unwrap(), file_id),
+
+        ParsedType::Assert(n) => unimplemented!(),
```

Comment:
We can keep it like this for now, but in the future `assert` can be implemented as a contract annotation, as it is done in tests.